src/Symfony/Component/Security/Core/Authorization/Voter/ClosureVoter.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+use Symfony\Component\Security\Core\Authentication\AuthenticationTrustResolverInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+
+/**
+ * This voter allows using a closure as the attribute being voted on.
+ * The following arguments are passed to the closure, in this order:
+ *
+ * - The token being used for voting
+ * - The subject of the vote
+ * - The access decision manager
+ * - The trust resolver
```

Comment:
Sorry if this has been discussed before but the ExpressionVoter gives access to $roleNames and to $authChecker - but not to $accessDecisionManager.
Shouldn't we do the same here?

Reply:
To summarize a bit [this thread](https://github.com/symfony/symfony/pull/59150#discussion_r1877891991), Mathias, Robin and Christophe would prefer to move away from role based access decisions to encourage attributes based decisions. This is why it was discouraged to give access to `$roleNames`.

About the `$authChecker` or `$accessDecisionManager`, stof raised a better compatiblity with what's already existing in [this comment](https://github.com/symfony/symfony/pull/59150#discussion_r1890041130). Does this answer your question?

Reply:
Giving access to the access decision manager instead of the auth checker is to make those callables compatible with the new feature allowing to vote on a different token than the current one. The AuthorizationChecker is about checking authorization for the current token only.

Reply:
If we go with [my proposal above](https://github.com/symfony/symfony/pull/59150#discussion_r1950531398), we can provide all at once.

Reply:
Anything using the $authChecker would be incompatible with the UserAuthorizationChecker (without any explicit failure: it would vote on a different token instead of the expected one, potentially create subtle bugs in projects). So I think this is a bad idea to make it available.

```diff
-$result = $authorizationChecker->isGranted($permission, $subject);
+$result = $accessDecisionManager->decide($token, [$permission], $subject);
```

Reply:
btw, now that the AuthorizationChecker does not support voting on multiple attributes at once, we might want to work on changing the type one level below in the AccessDecisionManager (which might then allow changing the signature in voters in the next cycle of major version as we would not rely anymore on the AccessDecisionManager passing an array to the voter)